kr8s/_async_utils.py

```diff
@@ -123,6 +113,28 @@ def run_sync_inner(*args: P.args, **kwargs: P.kwargs) -> T:
     raise TypeError(f"Expected coroutine function, got {coro.__class__.__name__}")
 
 
+def run_sync_gen(
```

Comment:
We may need to make these names a little easier to underatand. The term `gen` is used in a few other places in this project and I didn't immediately realised this was a sync wrapper for generators. 

This is also an interesting opportunity to rethink the name `run_sync`. Because it doesn't actually run the coroutine synchronously, it just wraps in in a synchronous function that handles dispatching and blocking until complete. You still need to call it. Maybe `as_sync` makes more sense then `run_sync`.

I wonder if `run_sync` => `as_sync_func` and `run_sync_gen` => `as_sync_generator` would be clearer?